obspy/io/nordic/ellipse.py

```diff
@@ -0,0 +1,504 @@
+            return cls(None, None, None, center)
+        # Sort eigenvalues in decreasing order
+        sort_indices = np.argsort(evals)[::-1]
+        # Select semi-major and semi-minor axes
+        a, b = np.sqrt(evals[sort_indices[0]]), np.sqrt(evals[sort_indices[1]])
+        # Calculate angle of semi-major axis
+        x_v1, y_v1 = evecs[:, sort_indices[0]]
+        if y_v1 == 0.:
+            theta = 90.
+        else:
+            theta = (np.degrees(np.arctan((x_v1) / (y_v1))) + 180) % 180
+        return cls(a, b, theta, center)
+
+    @classmethod
+    def from_uncerts(cls, x_err, y_err, c_xy, center=(0, 0)):
+        """Set error ellipse using Nordic epicenter uncertainties
+
+        Call as e=Ellipse.from_uncerts(x_err,y_err,c_xy,center)
+
+        :param x_err: x error (dist_units)
+        :type x_err: float
+        :param y_err: y error (dist_units)
+        :type y_err: float
+        :param c_xy:  x-y covariance (dist_units^2)
+        :type c_xy: float
+        :param center: center position (x,y)
+        :type center: 2-tuple of numeric
+        :return: ellipse
+        :rtype: :class: `~obspy.io.nordic.ellipse.Ellipse`
+        """
+        cov = [[x_err**2, c_xy], [c_xy, y_err**2]]
+        return cls.from_cov(cov, center)
+
+    @classmethod
+    def from_uncerts_baz(cls, x_err, y_err, c_xy, dist, baz,
+                         viewpoint=(0, 0)):
+        """Set error ellipse using uncertainties, distance and back-azimuth
+
+        Inputs:
+        :param x_err: x error (dist_units)
+        :type x_err: float
+        :param y_err: y error (dist_units)
+        :type y_err: float
+        :param c_xy:  x-y covariance (dist_units^2)
+        :type c_xy: float
+        :param dist:  distance of center from observer
+        :type dist: float
+        :param baz:   back-azimuth from observer (degrees)
+        :type baz: float
+        :param viewpoint: observer's position
+        :type viewpoint: 2-tuple of floats
+        :return: ellipse
+        :rtype: :class: `~obspy.io.nordic.ellipse.Ellipse`
+        """
+        x = viewpoint[0] + dist * np.sin(np.radians(baz))
+        y = viewpoint[1] + dist * np.cos(np.radians(baz))
+        return cls.from_uncerts(x_err, y_err, c_xy, (x, y))
+
+    def _almost_good_cov(cov):
+        """Checks if a covariance matrix is "almost good" (c_xy is greater
+        than c_xx and/or c_yy, but only by a little bit)
+        """
+        if cov[0][1] != cov[1][0]:
+            return False
+        ratio = cov[0][1]**2 / (cov[0][0] * cov[1][1])
+        if (ratio > 1) and (ratio < 1.1):
+            return True
+        return False
+
+    def _fix_cov(cov):
+        """Correct an "almost good" covariance matrix by making c_xx*c_yy
+        > c_xy^2
+        """
+        if not Ellipse._almost_good_cov(cov):
+            return None
+        ratio = cov[0][1]**2 / (cov[0][0] * cov[1][1])
+        cov[0][0] *= np.sqrt(ratio)*1.01
```

Comment:
Needs to have spaces around the `*` on this and the next line.

Reply:
Oops, missed that.  It's strange that flake8 on my machine doesn't catch that, but circlei does!